scripts/globals/battlefield.lua

```diff
@@ -68,10 +70,788 @@ xi.battlefield.dropChance =
+    end
+
+    -- Validate battlefield status
+    if player:hasStatusEffect(xi.effect.BATTLEFIELD) and not onUpdate then
+        player:messageBasic(xi.msg.basic.WAIT_LONGER, 0, 0)
+        return false
+    end
+
+    -- Check if another party member has battlefield status effect. If so, don't allow trade.
+    local alliance = player:getAlliance()
+    for _, member in pairs(alliance) do
+        if member:hasStatusEffect(xi.effect.BATTLEFIELD) then
+            player:messageBasic(xi.msg.basic.WAIT_LONGER, 0, 0)
+
+            return false
+        end
+    end
+
+    local zoneId = player:getZoneID()
+
+    -- Determine which battlefields are available given the traded items
+    local options = xi.battlefield.getBattlefieldOptions(player, npc, trade)
+    if options == 0 then
+        local noEntryMessage = zones[zoneId].text.NO_BATTLEFIELD_ENTRY
+        if noEntryMessage then
+            player:messageSpecial(noEntryMessage)
+        end
+        return false
+    end
+
+    -- Ensure that the traded item(s) are not worn out
+    local contents = xi.battlefield.contentsByZone[zoneId]
+    for _, content in ipairs(contents) do
+        if #content.requiredItems > 0 and content.requiredItems[1].wornMessage and player:hasWornItem(content.requiredItems[1]) then
+            player:messageBasic(content.requiredItems[1].wornMessage)
+            return false
+        end
+    end
+
+    if not onUpdate then
+        -- Open menu of valid battlefields
+        player:startEvent(32000, 0, 0, 0, options, 0, 0, 0, 0)
+    end
+
+    return true
+end
+
+function Battlefield.onEntryTrigger(player, npc)
+    -- Cannot enter if anyone in party is level/master sync'd
+    if xi.battlefield.rejectLevelSyncedParty(player, npc) then
+        return false
+    end
+
+    -- Player has battlefield status effect. That means a battlefield is open OR the player is inside a battlefield.
+    if player:hasStatusEffect(xi.effect.BATTLEFIELD) then
+        -- Player is outside battlefield. Attempting to enter.
+        local status = player:getStatusEffect(xi.effect.BATTLEFIELD)
+        local id = status:getPower()
+
+        local content = xi.battlefield.contents[id]
+        if not content then
+            return false
+        end
+
+        if not content:checkRequirements(player, npc, content.id, false) then
+            return false
+        end
+
+        player:startEvent(32000, 0, 0, 0, content.index, 0, 0, 0, 0)
+        return true
+    end
+
+    -- Player doesn't have battlefield status effect. That means player wants to register a new battlefield OR is attempting to enter a closed one.
+    -- Check if another party member has battlefield status effect. If so, that means the player is trying to enter a closed battlefield.
+    local alliance = player:getAlliance()
+    for _, member in pairs(alliance) do
+        if member:hasStatusEffect(xi.effect.BATTLEFIELD) then
+            -- player:messageSpecial() -- You are eligible but cannot enter.
```

Comment:
Is this commented code intentional?

Reply:
Kind of? I meant to put a TODO on it. Basically there needs to be some work done on fixing that messaging. The old system was missing a lot of messages.